### DISCUSSION TITLE
User Management

### DISCUSSION THREAD

**Discussion by milcaxx** [OctoFarm/OctoFarm#626] [Q&A] [answered]
Hello :)
I've been searching for quite a lot of time and i can't seem to find the information anywhere. 
Is there a User Management with Octofarm? Can you have multiple users and determine their rights like print with/without authorization from admin for example.
Thx for the help !!

**Comment by NotExpectedYet** [comment, marked as answer]
> Hello :)
> I've been searching for quite a lot of time and i can't seem to find the information anywhere.
> Is there a User Management with Octofarm? Can you have multiple users and determine their rights like print with/without authorization from admin for example.
> Thx for the help !!

It's coming at some point. Not sure when. Currently any user can start prints and you'd have to register the user yourself with that mechanism.  The first user becomes Administrator and they get permissions to add/remove and manage printers on the farm. Any subsequent user becomes a "user" permission and doesn't get access to those. Hope that helps. There's a feature request for this I believe.

**Reply by milcaxx** [reply]
Helps a lot!! thank you for the answer

**Reply by NotExpectedYet** [reply]
I've added this for V1.2 :) 

It's got CRUD user management. You can also reset the password for users.

**Reply by NotExpectedYet** [reply]
The permissions still need a lot of work but at the very least this is implemented aha

**Reply by NotExpectedYet** [reply]
![image](https://user-images.githubusercontent.com/61137988/165632398-da8d8931-1102-43b4-a0f7-bb420a14a022.png)

I've laid some more ground work for the further advancements mentions. There is a new print information screen that has the current user printing on it. 

![image](https://user-images.githubusercontent.com/61137988/165632548-2c4e1da6-f5da-4b91-ad6c-4a7974a311d8.png)


Admins can also see a actions log in the printer manager. It's currently only displaying the last 100 actions but there are more kept in the log, to be advanced further in V1.2+ 

I'm going to be doing some serious work on pushing the OctoPrint client commands back end, after 1.2 release, to help enable more granular control over this so it's actually possible to do all the good stuff like disabling print authorization on printers/file uploading and all the rest.
